source/signalzeug/source/ScopedConnection.cpp

```diff
@@ -14,15 +14,32 @@ ScopedConnection::ScopedConnection(const Connection & connection)
 {
 }
 
+ScopedConnection::ScopedConnection(ScopedConnection && other)
+:   m_connection(other.m_connection)
+{
+    other.m_connection = Connection();
+}
+
 ScopedConnection::~ScopedConnection()
 {
 	m_connection.disconnect();
 }
 
-void ScopedConnection::operator=(const Connection & connection)
+ScopedConnection& ScopedConnection::operator=(const Connection & connection)
 {
 	m_connection.disconnect();
 	m_connection = connection;
+
+    return *this;
+}
+
+ScopedConnection& ScopedConnection::operator=(ScopedConnection && other)
+{
+    m_connection.disconnect();
+    m_connection = other.m_connection;
```

Comment:
R-value references indicate only, that the resource may be moved from. We cannot assume that the object is destroyed afterwards, instead it must be left in "some valid but otherwise indeterminate state" (http://en.cppreference.com/w/cpp/language/move_operator).
Therefore, we should not rely on `other` to free the old connection at some point, but do it ourselves (this is more a general procedure, as we not are dealing with pointers here).